### PR TITLE
New Box

### PR DESCRIPTION
**Pull request by chrisnojima**
This introduces a new Box (box2, maybe we can think of a better name). This gives us some props to help with common layout issues.

I applied this to the devices-page since its simple. I also merged desktop and mobile since @cecileboucheron updated the design to use the <NameWithIcon> and made them both use the vertical layout

Box2 props:
```
export type Box2Props = {
  children: React.Node,
  direction: 'horizontal' | 'vertical',
  fullHeight?: boolean,
  fullWidth?: boolean,
  style?: StylesCrossPlatform,
  gap?: $Keys<typeof globalMargins>,
  gapStart?: boolean,
  gapEnd?: boolean,
}
```

- [x] New! Box2
- [x] Cleanup device page and use new design, unify desktop and mobile
- [x] Fixed up <Meta> component. Style typed and pulled color/backgroundColor into its own props
- [x] More storybook

### CODE REVIEW

File: shared/common-adapters/name-with-icon.js [hunk 1]
```diff
@@ -54,7 +54,6 @@ const NameWithIconVertical = (props: Props) => {
           <Icon
             type={props.icon || ''}
             style={{
-              fontSize: adapterProps.iconSize,
```
Comment (chrisnojima) [inline shared/common-adapters/name-with-icon.js]: causes a yellowbox. shouldn't be necessary

File: shared/devices/device-page/index.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+// @flow
+import * as React from 'react'
```
Comment (chrisnojima) [inline shared/devices/device-page/index.js]: unified and cleaned up

File: shared/profile/index.native.js [hunk 3]
```diff
@@ -82,7 +82,7 @@ const ShowcasedTeamRow = ({
       <Text style={{color: globalColors.black_75}} type="BodySemiboldLink">
         {team.fqName}
       </Text>
-      {team.open && <Meta style={styleMeta} title="OPEN" />}
+      {team.open && <Meta style={styleMeta} backgroundColor={globalColors.green} title="open" />}
```
Comment (chrisnojima) [inline shared/profile/index.native.js]: i changed title to lowercase so its more clear that component itself uppercases this stuff

File: shared/stories/index.native.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ const stories = {...commonStories, ...nativeStories}
 
 const scrollViewDecorator = story => [
   <StatusBar key="statusbar" hidden={true} />,
-  <ScrollView key="scrollview" style={{flex: 1}} contentContainerStyle={{height: '100%'}}>
```
Comment (chrisnojima) [inline shared/stories/index.native.js]: i think this is actually what we want

File: shared/stories/storybook.shared.js [hunk 5]
```diff
@@ -18,8 +18,11 @@ const unexpected = (name: string) => () => {
  *                       The Provider will ignore all dispatched actions. It also wraps the component
  *                       tree in an <ErrorBoundary /> that adds auxiliary info in case of an error.
  */
+// Redux doens't allow swapping the store given a single provider so we use a new key to force a new provider to
+// work around this issue
+let uniqueProviderKey = 1
 const createPropProvider = (map: SelectorMap) => (story: () => React.Node) => (
-  <Provider store={createStore(state => state, map)}>
+  <Provider key={`provider:${uniqueProviderKey++}`} store={createStore(state => state, map)}>
```
Comment (chrisnojima) [inline shared/stories/storybook.shared.js]: if you switch stories sometimes you'd get redboxes. Provider doesn't expect store to change dynamically and it has explicit errors about this, so as a workaround we entirely switch out the provider every time

**Comment by chrisnojima** [comment]
localhost_6006__selectedkind common selectedstory box2 full 0 addons 0 stories 0 panelright 0 addonpanel storybook 2factions 2factions-panel 3

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/common-adapters/box.native.js [hunk 6]
```diff
@@ -1,2 +1,71 @@
...
+import type {Box2Props} from './box'
+
+const Box = View
+
+const injectGaps = (Component, _children, gap, gapStart, gapEnd) => {
+  let children = _children
+  if (gap) {
+    children = intersperseFn(index => <Component key={index} gap={gap} />, React.Children.toArray(_children))
+    if (gapStart) {
+      children.unshift(<HBoxGap key="gapStart" gap={gap} />)
```
Comment (akalin-keybase) [inline shared/common-adapters/box.native.js]: this should be `Component` too right?

File: shared/common-adapters/box.native.js [hunk 7]
```diff
@@ -1,2 +1,71 @@
...
+    }
+    if (gapEnd) {
+      children.push(<HBoxGap key="gapEnd" gap={gap} />)
```
Comment (akalin-keybase) [inline shared/common-adapters/box.native.js]: here too

File: shared/common-adapters/box.native.js [hunk 8]
```diff
@@ -1,2 +1,71 @@
...
+import * as React from 'react'
+import {View} from 'react-native'
+import {globalStyles, collapseStyles, globalMargins} from '../styles'
+import {intersperseFn} from '../util/arrays'
+
+import type {Box2Props} from './box'
+
+const Box = View
+
+const injectGaps = (Component, _children, gap, gapStart, gapEnd) => {
```
Comment (akalin-keybase) [inline shared/common-adapters/box.native.js]: worth pulling this out into `box.shared.js` or something?

File: shared/stories/storybook.shared.js [hunk 9]
```diff
@@ -18,8 +18,11 @@ const unexpected = (name: string) => () => {
  *                       The Provider will ignore all dispatched actions. It also wraps the component
  *                       tree in an <ErrorBoundary /> that adds auxiliary info in case of an error.
  */
+// Redux doens't allow swapping the store given a single provider so we use a new key to force a new provider to
```
Comment (akalin-keybase) [inline shared/stories/storybook.shared.js]: doens't -> doesn't

File: shared/common-adapters/box.native.js (hunk 8 above)
Comment (chrisnojima) [reply shared/common-adapters/box.native.js]: i figured this is so low level and it'll barely change it wasn't above the threshold to do that yet, could go either way

**Pull request merged**